### PR TITLE
Allow Lua file preview in asset browser

### PR DESCRIPTION
**Pull request by tjmgd**
Signed-off-by: T.J. McGrath-Daly <tj.mcgrath.daly@huawei.com>

### CODE REVIEW

File: Gems/LuaScript/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+# Description:
+# Create: 2021-07-014
+
```
Comment (lsemp3d) [inline Gems/LuaScript/CMakeLists.txt:4]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Gems/LuaScript/Code/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+# Description: build Terrain and Terrain.Static.
+# Create: 2021-07-14
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/CMakeLists.txt:3]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Gems/LuaScript/Code/Source/LuaScriptModule.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: LuaScriptModule.
+ * Create: 2021-07-14
+ */
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/Source/LuaScriptModule.cpp:5]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Gems/LuaScript/Code/Source/LuaScriptSystemComponent.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: LuaScriptSystemComponent.
+ * Create: 2021-07-14
+ */
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/Source/LuaScriptSystemComponent.h:5]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Gems/LuaScript/Code/Source/Previewer/LuaScriptPreviewer.cpp [hunk 5]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: LuaScriptPreviewer.
+ * Create: 2021-07-14
+ */
+
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/Source/Previewer/LuaScriptPreviewer.cpp:6]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Gems/LuaScript/Code/Source/Previewer/LuaScriptPreviewer.h [hunk 6]
```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: LuaScriptPreviewer.
+ * Create: 2021-07-14
+ */
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/Source/Previewer/LuaScriptPreviewer.h:5]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Gems/LuaScript/Code/Source/Previewer/LuaScriptPreviewerFactory.cpp [hunk 7]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: LuaScriptPreviewerFactory.
+ * Create: 2021-07-14
+ */
+
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/Source/Previewer/LuaScriptPreviewerFactory.cpp:6]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Gems/LuaScript/Code/Source/Previewer/LuaScriptPreviewerFactory.h [hunk 8]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: LuaScriptPreviewerFactory.
+ * Create: 2021-07-14
+ */
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/Source/Previewer/LuaScriptPreviewerFactory.h:5]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Gems/LuaScript/Code/Source/Previewer/ui_LuaScriptPreviewer.h [hunk 9]
```diff
@@ -0,0 +1,86 @@
+/********************************************************************************
+** Form generated from reading UI file 'LuaScriptPreviewer.ui'
+**
+** Created by: Qt User Interface Compiler version 6.1.2
+**
+** WARNING! All changes made in this file will be lost when recompiling UI file!
+********************************************************************************/
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/Source/Previewer/ui_LuaScriptPreviewer.h:7]: This file is generated by the Qt UIC tool through cmake, it does not need to be committed to source control

File: Gems/LuaScript/Code/luascript_editor_files.cmake [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+# Description: luascript_editor_files.
+# Create: 2021-07-14
+
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/luascript_editor_files.cmake:4]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Gems/LuaScript/Code/luascript_files.cmake [hunk 11]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+# Description: luascript_files.
+# Create: 2021-07-14
+
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/luascript_files.cmake:4]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Gems/LuaScript/Code/luascript_shared_files.cmake [hunk 12]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+# Description: luascript_shared_files.
+# Create: 2021-07-14
+
```
Comment (lsemp3d) [inline Gems/LuaScript/Code/luascript_shared_files.cmake:4]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: engine.json [hunk 13]
```diff
@@ -84,9 +84,11 @@
         "Gems/Vegetation",
         "Gems/VideoPlaybackFramework",
         "Gems/VirtualGamepad",
-        "Gems/WhiteBox"
+        "Gems/WhiteBox",
+        "Gems/LuaScript"
     ],
     "projects": [
+        "UCDDemo",
```
Comment (lsemp3d) [inline engine.json:91]: This should not be merged in

File: Code/Framework/AzCore/AzCore/Script/ScriptAdapterBus.h [hunk 14]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: AdapterEvents.
+ * Create: 2021-06-28
+ */
```
Comment (lsemp3d) [inline Code/Framework/AzCore/AzCore/Script/ScriptAdapterBus.h:5]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.cpp [hunk 15]
```diff
@@ -0,0 +1,475 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: ScriptAdapterComponent.
+ * Create: 2021-06-28
+ */
```
Comment (lsemp3d) [inline Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.cpp:5]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.h [hunk 16]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: ScriptAdapterComponent.
+ * Create: 2021-06-28
+ */
```
Comment (lsemp3d) [inline Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.h:5]: Please see: 
https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#copyright-header-requirements
for copyright header requirements

File: Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.cpp [hunk 17]
```diff
@@ -0,0 +1,475 @@
...
+
+    void ScriptAdapterComponent::Deactivate()
+    {
+        AdapterBus::Handler::BusDisconnect();
+    }
+
+    AZStd::string ScriptAdapterComponent::GetProjectPath()
+    {
+        ComponentApplication* app = nullptr;
+        EBUS_EVENT_RESULT(app, ComponentApplicationBus, GetApplication);
```
Comment (lsemp3d) [inline Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.cpp:51]: The ebus macro is considered the legacy syntax, it's recommended to use the following syntax:
`ComponentApplicationBus::BroadcastResult(app, &ComponentApplication::GetApplication);`

File: Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.cpp [hunk 18]
```diff
@@ -0,0 +1,475 @@
...
+        {
+            dacl.AddDeniedAce(sid, DELETE | FILE_DELETE_CHILD | FILE_WRITE_DATA | FILE_APPEND_DATA | FILE_WRITE_EA | FILE_WRITE_ATTRIBUTES, OBJECT_INHERIT_ACE | CONTAINER_INHERIT_ACE);
+        }
+
+        return AtlSetDacl(genPath.c_str(), SE_FILE_OBJECT, dacl) ? true : false;
+    #elif defined(LINUX)
+        SetGenDirWritableRecursive(GetGenDirPath().c_str(), isWritable);
+        return true;
+    #endif
+    }
```
Comment (lsemp3d) [inline Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.cpp:99]: This code needs to be modified to use the platform abstraction layer (PAL)

https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#code

Code
Required: All platform specific code needs to be implemented according to the patterns defined by the PAL effort (Platform Abstraction Layer).

Avoid adding platform specific defines to the codebase.

**Comment by tjmgd** [comment]
Give the number of problems here I am closing the PR until they can be resolved.

**Pull request closed without merging**